python/tests/test_dict_encoding.py

```diff
@@ -404,3 +557,51 @@ def test_provenances(self):
             "provenances",
             ["record", "record_offset", "timestamp", "timestamp_offset"],
         )
+
+    def test_top_level_metadata(self):
+        tables = get_example_tables()
+        d = tables.asdict()
+        # None should give default value
+        d["metadata"] = None
+        lwt = c_module.LightweightTableCollection()
+        lwt.fromdict(d)
+        out = lwt.asdict()
+        self.assertNotIn("metadata", out)
+        tables = tskit.TableCollection.fromdict(out)
+        # We only removed the metadata, not the schema. So empty bytefield
+        # still gets interpreted
+        self.assertEqual(tables.metadata, {"top-level": []})
+        # Missing is tested in TestMissingData above
+
+    def test_top_level_metadata_schema(self):
+        tables = get_example_tables()
+        d = tables.asdict()
+        # None should give default value
+        d["metadata_schema"] = None
+        lwt = c_module.LightweightTableCollection()
+        lwt.fromdict(d)
+        out = lwt.asdict()
+        self.assertNotIn("metadata_schema", out)
+        tables = tskit.TableCollection.fromdict(out)
+        self.assertEqual(str(tables.metadata_schema), "")
+        # Missing is tested in TestMissingData above
+
+
+class TestExamples(unittest.TestCase):
+    def test_pickled_examples(self):
+        seen_current_msprime = False
+        test_dir = pathlib.Path(__file__).parent / "data/dict-encodings"
+        for filename in os.listdir(test_dir):
+            if "pkl" not in filename:
+                continue
+            if (
+                "msprime" in filename
+                and filename.split("-")[1] == f"{msprime.__version__}.pkl"
+            ):
+                seen_current_msprime = True
+            with open(test_dir / filename, "rb") as f:
+                d = pickle.load(f)
+                lwt = c_module.LightweightTableCollection()
+                lwt.fromdict(d)
+                tskit.TableCollection.fromdict(d)
+        self.assertTrue(seen_current_msprime)
```

Comment:
This is going to break for some developers who install development versions of msprime. I don't think it gains us enough to make the hassle worth it, so I'd say just read in what's there and maybe assert that at least one file has been read (disturbingly easy for tests like this to fall through without doing anything).

Reply:
Also, we'd have to update this for every release of msprime - hardly seems worth it, since the dict encoding is going to change extremely rarely.

Reply:
Yeah, this was wrong! Fixed in https://github.com/tskit-dev/tskit/pull/687/commits/a8d83651c0842dacfbb4605ca067c82efbec1831